File: hw/basurto/u1/hw04Arrays/Python/view/chicken_farm_simulator.py
from model.chicken import Chicken
from datetime import date

chickens = []
option = 'y'

while option == 'y' or option == 'Y':

    print("\n--- ENTER CHICKEN DATA ---")

    id = int(input("Id: "))
    name = input("Name: ")
    color = input("Color: ")

    year = int(input("Year of birth: "))
    month = int(input("Month (1-12): "))
    day = int(input("Day: "))

    born_on_date = date(year, month, day)

    age = int(input("Age: "))

    is_molting_input = input("Is molting? (true/false): ").lower()
    is_molting = is_molting_input == "true"

    chicken = Chicken(
        id,
        name,
        color,
        born_on_date,
        age,
        is_molting
    )

    chickens.append(chicken)

    option = input("Do you want to add another chicken? (y/n): ")

print("\n--- CHICKENS LIST ---")

for i in range(len(chickens)):
    print(f"chicken [{i + 1}] --> {chickens[i]}")

File: hw/basurto/u1/hw04Arrays/JavaScript/view/ChickenFarmSimulator.js
import promptSync from "prompt-sync";
import Chicken from "../model/Chicken.js";

const prompt = promptSync();

let chickens = [];
let option = "y";

while (option.toLowerCase() === "y") {

    console.log("\n--- ENTER CHICKEN DATA ---");

    let id = Number(prompt("Id: "));
    let name = prompt("Name: ");
    let color = prompt("Color: ");

    let year = Number(prompt("Year of birth: "));
    let month = Number(prompt("Month (1-12): "));
    let day = Number(prompt("Day: "));

    let bornOnDate = new Date(year, month - 1, day);

    let age = Number(prompt("Age: "));

    let isMoltingInput = prompt("Is molting? (true/false): ").toLowerCase();
    let isMolting = isMoltingInput === "true";

    let chicken = new Chicken(
        id,
        name,
        color,
        bornOnDate,
        age,
        isMolting
    );

    chickens.push(chicken);

    option = prompt("Do you want to add another chicken? (y/n): ");
}

console.log("\n--- CHICKENS LIST ---");

for (let i = 0; i < chickens.length; i++) {
    console.log(`chicken [${i + 1}] --> ${chickens[i].toString()}`);
}